Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -20,20 +20,14 @@
             HOMEBREW_LINUX_DEFAULT_PREFIX].include?(prefix) %>
     prefix "<%= prefix %>"
     <% end %>
-    <% if cellar.is_a? Symbol %>
-    cellar :<%= cellar %>
-    <% elsif ![Homebrew::DEFAULT_MACOS_CELLAR,
-               Homebrew::DEFAULT_MACOS_ARM_CELLAR,
-               Homebrew::DEFAULT_LINUX_CELLAR].include?(cellar) %>
-    cellar "<%= cellar %>"
-    <% end %>
     <% if rebuild.positive? %>
     rebuild <%= rebuild %>
     <% end %>
     <% checksums.each do |checksum_type, checksum_values| %>
     <% checksum_values.each do |checksum_value| %>
-    <% checksum, macos = checksum_value.shift %>
-    <%= checksum_type %> "<%= checksum %>" => :<%= macos %>
+    <% hexdigest, tag = checksum_value.first %>
+    <% cellar = checksum_value["cellar"] %>
+    <%= checksum_type %> "<%= hexdigest %>" => :<%= tag %><% if cellar.is_a? Symbol %>, "cellar" => :<%= cellar %><% elsif ![Homebrew::DEFAULT_MACOS_CELLAR, Homebrew::DEFAULT_MACOS_ARM_CELLAR, Homebrew::DEFAULT_LINUX_CELLAR].include?(cellar) %>, "cellar" => "<%= cellar %>"<% end %>
```

Comment:
This line needs to be shorter. Pull logic out of ERB if needed.

Reply:
I'll rework this in the second PR for the writing part.